pilot/pkg/model/push_context.go

```diff
@@ -1605,7 +1605,7 @@ func (ps *PushContext) initEnvoyFilters(env *Environment) error {
 		return err
 	}
 
-	sortConfigByCreationTime(envoyFilterConfigs)
+	sort.SliceStable(envoyFilterConfigs, func(i, j int) bool { return envoyFilterConfigs[i].Name < envoyFilterConfigs[j].Name })
```

Comment:
Since this is replacing one type of sorting with another it will be a breaking change in behaviour.
Is it possible to restrict it with some type of opt-in?

otherwise having a new priority field which defaults in the middle (unset = middle priority) then if. Then this sort will be by priority first and then by creation time like before.

@kyessenov ?

Reply:
Yes. I can do opt-in. We can remove/change in opt-in when we decide on priority field in API.

Reply:
That makes sense, do opt-in for now and resolve API question concurrently.